### PR TITLE
documents member

### PR DESCRIPTION
**Pull request by jmgomez**

### CODE REVIEW

File: doc/manual_experimental.md [hunk 1]
```diff
@@ -2381,3 +2381,60 @@ proc makeCppClass(): NimClass {.constructor: "NimClass() : CppClass(0, 0)".} =
 In the example above `CppClass` has a deleted default constructor. Notice how by using the constructor syntax, one can call the appropiate constructor.
 
 Notice when calling a constructor in the section of a global variable initialization, it will be called before `NimMain` meaning Nim is not fully initialized.
+
+Member pragma
+=============
+
+In the same vein as the `constructor` and `virtual` pragmas, the `member` pragma can be used to attatch a proc to a type in C++. It's more flexible than `virtual` in the sense that it accepts not only names but also operators or destructors.
```
Comment (juancarlospaco) [inline doc/manual_experimental.md]:
```suggestion
Similar to the `constructor` and `virtual` pragmas, the `member` pragma can be used to attach a `proc` or `func` to a type in C++. 
It is more flexible than `virtual` in the sense that it accepts not only names but also operators or destructors.
```

"attatch" typo, also simple words help non-native speakers ("in the veins" etc).

File: doc/manual_experimental.md [hunk 2]
```diff
@@ -2381,3 +2381,60 @@ proc makeCppClass(): NimClass {.constructor: "NimClass() : CppClass(0, 0)".} =
...
+2
+false
+destructing
+destructing
+```
+
+Notice how the C++ destructor is called automatically. Also notice the double implementation of `==` as an operator in Nim but also in C++. This is useful if you need the type to match some C++ `concept` or `trait` when interoping. 
+
+A side effect of being able to declare C++ operators, is that you can now also create a
+C++ functor to have seamless interop with C++ lambdas (sysntactic sugar for functors).
```
Comment (juancarlospaco) [inline doc/manual_experimental.md]:
```suggestion
C++ functor to have seamless interop with C++ lambdas (syntactic sugar for functors).
```

File: doc/manual_experimental.md [hunk 3]
```diff
@@ -2381,3 +2381,60 @@ proc makeCppClass(): NimClass {.constructor: "NimClass() : CppClass(0, 0)".} =
...
+
+For example,
```
Comment (juancarlospaco) [inline doc/manual_experimental.md]:
```suggestion
For example:
```

File: doc/manual_experimental.md [hunk 4]
```diff
@@ -2381,3 +2381,60 @@ proc makeCppClass(): NimClass {.constructor: "NimClass() : CppClass(0, 0)".} =
...
+  NimFunctor = object
+    discard
+proc invoke(f: NimFunctor, n:int) {.member:"operator ()('2 #2)" .} = 
+  echo "FunctorSupport!"
+
+{.experimental: "callOperator".}
+proc `()`(f: NimFunctor, n:int) {.importcpp:"#(@)" .} 
+NimFunctor()(1)
+```
+Notice we are also using the overload of `()` to have the same semantincs in Nim, but on the `importcpp` we are importing the functor as a function. This allows to easy interop with functions that accepts for example a `const` operator in its signature.
```
Comment (juancarlospaco) [inline doc/manual_experimental.md]:
```suggestion
Notice we are also using the overload of `()` to have the same semantics in Nim, but on the `importcpp` we are importing the functor as a function. 
This allows to easy interop with functions that accepts for example a `const` operator in its signature. 
```

File: doc/manual_experimental.md [hunk 5]
```diff
@@ -2381,3 +2381,62 @@ proc makeCppClass(): NimClass {.constructor: "NimClass() : CppClass(0, 0)".} =
...
+Member pragma
+=============
+
+Similar to the `constructor` and `virtual` pragmas, the `member` pragma can be used to attach a `proc` or `func` to a type in C++. 
+It is more flexible than `virtual` in the sense that it accepts not only names but also operators or destructors.
+
+For example:
+
+```nim
+proc print(s: cstring) {.importcpp:"printf(@)", header:"<stdio.h>".}
```
Comment (Araq) [inline doc/manual_experimental.md]:
```suggestion
proc print(s: cstring) {.importcpp: "printf(@)", header: "<stdio.h>".}
```

File: doc/manual_experimental.md [hunk 6]
```diff
@@ -2381,3 +2381,62 @@ proc makeCppClass(): NimClass {.constructor: "NimClass() : CppClass(0, 0)".} =
...
+
+type
+  Doo  {.exportc.} = object
```
Comment (Araq) [inline doc/manual_experimental.md]:
```suggestion
  Doo {.exportc.} = object
```

File: doc/manual_experimental.md [hunk 7]
```diff
@@ -2381,3 +2381,62 @@ proc makeCppClass(): NimClass {.constructor: "NimClass() : CppClass(0, 0)".} =
...
+    test: int
+
+proc memberProc(f: Doo) {.member.} = 
+  echo $f.test
+
+proc destructor(f: Doo) {.member: "~'1()", used.} = 
+  print "destructing\n"
+
+proc `==`(self, other: Doo): bool {.member:"operator==('2 const & #2) const -> '0"} =
```
Comment (Araq) [inline doc/manual_experimental.md]:
```suggestion
proc `==`(self, other: Doo): bool {.member: "operator==('2 const & #2) const -> '0".} = 
```

File: doc/manual_experimental.md [hunk 8]
```diff
@@ -2381,3 +2381,62 @@ proc makeCppClass(): NimClass {.constructor: "NimClass() : CppClass(0, 0)".} =
...
+A side effect of being able to declare C++ operators, is that you can now also create a
+C++ functor to have seamless interop with C++ lambdas (syntactic sugar for functors).
+
+For example:
+
+```nim
+type
+  NimFunctor = object
+    discard
+proc invoke(f: NimFunctor, n:int) {.member:"operator ()('2 #2)" .} =
```
Comment (Araq) [inline doc/manual_experimental.md]:
```suggestion
proc invoke(f: NimFunctor; n: int) {.member: "operator ()('2 #2)".} = 
```

File: doc/manual_experimental.md [hunk 9]
```diff
@@ -2381,3 +2381,62 @@ proc makeCppClass(): NimClass {.constructor: "NimClass() : CppClass(0, 0)".} =
...
+
+```nim
+type
+  NimFunctor = object
+    discard
+proc invoke(f: NimFunctor, n:int) {.member:"operator ()('2 #2)" .} = 
+  echo "FunctorSupport!"
+
+{.experimental: "callOperator".}
+proc `()`(f: NimFunctor, n:int) {.importcpp:"#(@)" .}
```
Comment (Araq) [inline doc/manual_experimental.md]:
```suggestion
proc `()`(f: NimFunctor; n:int) {.importcpp: "#(@)" .} 
```

File: doc/manual_experimental.md [hunk 10]
```diff
@@ -2381,3 +2381,62 @@ proc makeCppClass(): NimClass {.constructor: "NimClass() : CppClass(0, 0)".} =
...
+  NimFunctor = object
+    discard
+proc invoke(f: NimFunctor, n:int) {.member:"operator ()('2 #2)" .} = 
+  echo "FunctorSupport!"
+
+{.experimental: "callOperator".}
+proc `()`(f: NimFunctor, n:int) {.importcpp:"#(@)" .} 
+NimFunctor()(1)
+```
+Notice we are also using the overload of `()` to have the same semantics in Nim, but on the `importcpp` we are importing the functor as a function.
```
Comment (Araq) [inline doc/manual_experimental.md]:
```suggestion
Notice we use the overload of `()` to have the same semantics in Nim, but on the `importcpp` we import the functor as a function. 
```

**Pull request merged**